### PR TITLE
Update Java tooling documentation

### LINKED ISSUE #282: [BUG] Validation documentation is out of date
**Describe the bug**
The developer [documentation](https://github.com/eclipse-esmf/esmf-sdk/blob/main/documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc#validating-aspect-models) for validating Aspect Models still uses the deprecated validation API. It should be updated and describe in necessary detail the changes introduced in #244.

### PR DESCRIPTION
**Pull request by atextor**
Fixes #282.

This PR not only updates all Java tooling to reflect the current state of the
Java APIs, but it also changes the build setup: The documentation/ directory is
now both the root directory for the Antora documentation module, but is now at
the same time a valid Maven submodule. The
developer-guide/modules/tooling-guide/examples directory is configured as the
module's test-sources directory (and still serves as [Antora examples
directory](https://docs.antora.org/antora/latest/page/examples/)). Building jar,
deployment and installation plugins are disabled for the documentation module so
that it's only included in compilation and test execution.
This should prevent diverging of APIs and documentation in the future.

### CODE REVIEW

File: documentation/developer-guide/modules/tooling-guide/examples/GenerateJavaPojo.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      // Aspect as created by the AspectModelLoader
+      final Aspect aspect = // ...
+            // end::generate[]
+            // exclude the actual loading from the example to reduce noise
+            AspectModelResolver.loadAndResolveModel( modelFile ).flatMap( AspectModelLoader::getSingleAspect ).get();
+      // tag::generate[]
+
+      final JavaCodeGenerationConfig config = JavaCodeGenerationConfigBuilder.builder()
+            .enableJacksonAnnotations( true )
+            .packageName( "com.example.mycompany" ) // if left out, package is take from Aspect's namespace
```
Comment (RaMisess) [inline documentation/developer-guide/modules/tooling-guide/examples/GenerateJavaPojo.java:43]: package is taken

File: documentation/developer-guide/modules/tooling-guide/examples/GenerateJavaStaticClass.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      // Aspect as created by the AspectModelLoader
+      final Aspect aspect = // ...
+            // end::generate[]
+            // exclude the actual loading from the example to reduce noise
+            AspectModelResolver.loadAndResolveModel( modelFile ).flatMap( AspectModelLoader::getSingleAspect ).get();
+      // tag::generate[]
+
+      final JavaCodeGenerationConfig config = JavaCodeGenerationConfigBuilder.builder()
+            .enableJacksonAnnotations( true )
+            .packageName( "com.example.mycompany" ) // if left out, package is take from Aspect's namespace
```
Comment (RaMisess) [inline documentation/developer-guide/modules/tooling-guide/examples/GenerateJavaStaticClass.java:43]: typo: package is taken

File: documentation/developer-guide/modules/tooling-guide/examples/GenerateOpenApi.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            parameter:
+              name: parameter
+              in: path
+              description: "A parameter."
+              required: true
+              schema:
+                type: string
+            """ );
+
+      // Should the query API shall be added to the generated specification?
```
Comment (RaMisess) [inline documentation/developer-guide/modules/tooling-guide/examples/GenerateOpenApi.java:63]: Typo: should ... shall ....

File: documentation/developer-guide/modules/tooling-guide/examples/GenerateOpenApi.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                "description": "The ID of the unit.",
+                "required": true,
+                "schema": {
+                  "type": "string"
+                }
+              }
+            }
+            """ ) );
+
+      // Should the query API shall be added to the generated specification?
```
Comment (RaMisess) [inline documentation/developer-guide/modules/tooling-guide/examples/GenerateOpenApi.java:120]: typo: should ... shall ....

File: documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc [hunk 5]
```diff
@@ -142,9 +164,54 @@ examples for the different variants.
...
+<1> The file must be located in a directory according to the
+xref:tooling-guide:bamm-cli.adoc#models-directory-structure[models directory] structure.
+<2> You can check if loading succeeded using `versionedModel.isSuccess()` and
+`versionedModel.isFailure()`. If loading fails, `versionedModel.getCause()` provides more
+information.
+<3> The code inside `forEach` is only executed if the model could be loaded successfully.
+<4> List all statements in the RDF model that declare a `bamm:Aspect`.
+
+If you want to choose which model resolution strategy is used, you use the `resolveAspectModel()`
+method of the `AspectModelResolver`. In this case, you need to provide the resolution strategy and
```
Comment (RaMisess) [inline documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc:195]: typo: and and

File: documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc [hunk 6]
```diff
@@ -177,9 +277,41 @@ include::sds-developer-guide:ROOT:partial$sds-aspect-model-serializer-artifact.a
...
+++++
+</details>
+++++
+
+[source,java,indent=0,subs="+macros,+quotes"]
+----
+include::example$ValidateAspectModel.java[tags=validate]
+----
+
+<1> For an immediate human-readable result of the validation result, use the `ViolationFormatter` or
```
Comment (RaMisess) [inline documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc:308]: ... result of the validation result ... sounds a little strange, is there a better formulation?

File: documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc [hunk 7]
```diff
@@ -235,15 +405,45 @@ xref:bamm-specification:ROOT:payloads.adoc#mapping-to-json[mapping rules] as def
...
+
+[source,java,indent=0,subs="+macros,+quotes"]
+----
+include::example$GenerateJsonPayload.java[tags=generate]
+----
 
 === Generating JSON Schema
 
 The JSON schema generator creates a https://json-schema.org/[JSON Schema] that describes the payload for an Aspect model
 as it could be returned by an Aspect that implements that Aspect model. Currently, the generated schema corresponds to
```
Comment (RaMisess) [inline documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc:428]: corresponds to ???? 
-> isn't something missing here? (the include has been removed)

Comment (atextor) [reply documentation/developer-guide/modules/tooling-guide/pages/java-aspect-tooling.adoc:428]: This was actually broken before as well:
image

**Review by RaMisess — APPROVED** [review]

**Review by michelu89 — APPROVED** [review]

**Pull request merged**